bundle/src/main/java/com/adobe/acs/commons/models/injectors/impl/SharedValueMapValueInjector.java

```diff
@@ -21,106 +21,234 @@
+    public Object getValue(Object adaptable, String name, Type declaredType,
+                           AnnotatedElement element, DisposalCallbackRegistry callbackRegistry) {
         // sanity check
         if (element.getAnnotation(SharedValueMapValue.class) == null) {
             return null;
         }
 
-        Resource resource = getResource(adaptable);
-
-        if (resource != null) {
-            String rootPagePath = pageRootProvider.getRootPagePath(resource.getPath());
-            if (StringUtils.isNotBlank(rootPagePath)) {
-                ValueMap valueMap = null;
-                switch (element.getAnnotation(SharedValueMapValue.class).type()) {
-                    case MERGED:
-                        valueMap = getMergedProperties(rootPagePath, resource);
-                        break;
-                    case SHARED:
-                        valueMap = getSharedProperties(rootPagePath, resource);
-                        break;
-                    case GLOBAL:
-                        valueMap = getGlobalProperties(rootPagePath, resource);
-                        break;
-                    default:
-                        break;
-                }
-                if (valueMap != null) {
-                    return ReflectionUtil.convertValueMapValue(valueMap, name, declaredType);
-                }
+        final ValueMap valueMap;
+        if (sharedComponentProperties != null) {
+            valueMap = getValueMap(adaptable, element);
+        } else if (getValueType(element) == SharedComponentProperties.ValueTypes.MERGED) {
+            valueMap = Optional.ofNullable(getResource(adaptable))
+                    .map(Resource::getValueMap)
+                    .orElse(ValueMap.EMPTY);
+        } else {
+            valueMap = ValueMap.EMPTY;
+        }
+
+        return ReflectionUtil.convertValueMapValue(valueMap, name, declaredType);
+    }
+
+    private ValueMap getValueMap(Object adaptable, AnnotatedElement element) {
+        final SharedComponentProperties.ValueTypes valueType = getValueType(element);
+        if (valueType == null) {
+            return ValueMap.EMPTY;
+        }
+
+        final Resource resource = getResource(adaptable);
+        // we always need a resource, if only to determine if cached global properties are valid for the resource path
+        if (resource == null) {
+            return ValueMap.EMPTY;
+        }
+
+        // the root page path is used to test the validity of bindings
+        final String rootPagePath = sharedComponentProperties.getSharedPropertiesPagePath(resource);
+        if (rootPagePath == null) {
+            // when we have a resource but no root page path, we can at least satisfy MERGED with the resource valuemap
+            return valueType == SharedComponentProperties.ValueTypes.MERGED ? resource.getValueMap() : null;
```

Comment:
edge case bug here, where SCP is configured but there is no page root - the `else` case returns `null` but needs to instead return `ValueMap.EMPTY` if we dont want model objects pulling `GLOBAL` or `SHARED` `valueType` to blow up.